File: selfstudy/golang/books_filterlow.go
/*
task 256:
Books - FilterLow
You need to return only the values less than 50 from book page counts
*/

package main

import (
	"fmt"
	"reflect"
)

func Below_fiftyBooks(pages []int) []int {
	var res []int
	for _, v := range pages {
		if v < 50 { res = append(res, v) }
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Below_fiftyBooks([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Below_fiftyBooks([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Below_fiftyBooks([]int{1, 2, 3})},
		{[]int{100, 200}, Below_fiftyBooks([]int{100, 200})},
		{[]int{}, Below_fiftyBooks([]int{})},
	}
	
	for idx, t := range tests {
		res := Below_fiftyBooks(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works

File: selfstudy/python/books_filterlow.py
'''
task 156:
Books - FilterLow
You need to return only the values less than 50 from book page counts
'''

def below_fifty_books(pages):
    return [x for x in pages if x < 50]

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], below_fifty_books([10, 20, 30])),
        ([50, 60, 70], below_fifty_books([50, 60, 70])),
        ([1, 2, 3], below_fifty_books([1, 2, 3])),
        ([100, 200], below_fifty_books([100, 200])),
        ([], below_fifty_books([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = below_fifty_books(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# finally works
